src/msg/async/rdma/Infiniband.h

```diff
@@ -116,7 +116,11 @@ class DeviceList {
       delete devices[i];
     }
     delete []devices;
+    devices = NULL;
```

Comment:
no need to reset the member variable in destructor. as it won't be read anymore.

Reply:
@tchaikov Is it a good behaviour to let the stack/head use uncertain value in the process?

Reply:
why do you think we would read a destroyed object?

Reply:
what does "head" stand for?

Reply:
typo "head/heap".
I'll remove the un-necessary the assignment here. 
For the "read a destroyed object", it's my previous project debug experience. I won't use them in here now.